pie/default_settings.json

```diff
@@ -124,10 +124,16 @@
   "optimizer": "Adam", // optimizer type
   "clip_norm": 5.0, // clip norm of gradient up to this value
   "lr": 0.001,
-  "lr_factor": 0.75, // lr schedule (decrease lr by this factor after `lr_patience` epochs
-  // without improvement on dev-set data)
-  "min_lr": 0.000001, // minimum learning rate
-  "lr_patience": 2, // patience for lr schedule
+
+  "lr_scheduler": "ReduceLROnPlateau",
+  "lr_scheduler_params": {
```

Comment:
I definitely like this part.

Reply:
But, as I mentioned in my other PR, this basically breaks when initiating something else than a `Adam`.

Another option would be to have

```json
{
"lr_scheduler_params": {
  "Adam": {
    "mode": "max", ...
    }
  }
}
``` 

and this way provide defaults for things we tested ?

Reply:
But those parameters are not specific from the optimizer but from the scheduler, right? I have made some changes that should address this issue...

Reply:
I am sorry, I meant scheduler. I am a bit tired right now ^^. So I corrected the comment up there.